addons/test_mail/tests/test_mail_channel.py

```diff
@@ -245,6 +245,27 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(test_channel_group.channel_partner_ids, self.env['res.partner'])
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_multi_company_chat(self):
+        company_A = self.env['res.company'].create({'name': 'Company A'})
+
+        test_user = self.env['res.users'].create({
+            'login': 'nomail',
+            'name': 'No Mail',
+            'company_ids': [(6, 0, company_A.ids)],
+            'company_id': company_A.id
+        })
+        test_chat = self.env['mail.channel'].with_context(self._test_context).create({
+            'channel_partner_ids': [(4, test_user.partner_id.id), (4, self.user_admin.partner_id.id)],
+            'public': 'private',
+            'channel_type': 'chat',
+            'name': 'test'
+        })
+        try:
+            notification = test_chat.with_context(allowed_company_ids=self.user_admin.company_ids.ids)._channel_channel_notifications(test_user.partner_id.ids)
+        except AccessError:
+            notification = False
```

Comment:
Don't need for a dummy try/except here, just let the test crash if it doesn't work, the error message will be more clear.